service/geopmdpy/system_files.py

```diff
@@ -117,7 +117,7 @@ def secure_make_dirs(path, perm_mode=0o700):
             os.mkdir(path, mode=perm_mode)
     # If the path doesn't exist
     else:
-        os.mkdir(path, mode=perm_mode)
+        os.makedirs(path, mode=perm_mode)
```

Comment:
In the future, it would be nice to have a test that tries to create a path with an intermediate directory to prove that this all works.

Reply:
There is an issue here. The os.makedirs() call only applies the perm_mode to the leaf directory.  It would be good indeed to have a unit test that shows that the permissions are set correctly on intermediate directories as well.  The recommended solution in the documentation is to set the umask first:

https://docs.python.org/3/library/os.html#os.makedirs